src/components/ui/Utilities/formatDate.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import memoize from 'fast-memoize';
+
+export interface DateOptions {
+  weekday?: string;
+  year?: string;
+  month?: string;
+  day?: string;
+}
+
+const formatDateUnmemoized = (dateStr: string, locale: string | undefined, dateOptions?: DateOptions, todayText?: string, yesterdayText?: string) => {
```

Comment:
if passed local is 'undefined' then calling methods on Date object will throw exeption!

Reply:
It works as `undefined`. In fact, that is how we are calling it on the DateHeader story.